### PR TITLE
Refactor test API and include serial/multiserial tests

### PR DESCRIPTION
**Pull request by Acuadros95**

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/222/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/241/)

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,7 +31,7 @@ project(microxrcedds VERSION "2.0.0")
 
 set(_client_tag develop)
 set(_client_version 2.0.0)
-set(_agent_tag develop)
+set(_agent_tag feature/multiserial)
```
Comment (Acuadros95) [inline CMakeLists.txt:34]:
```suggestion
set(_agent_tag developl)
```
Modify before merge

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/223/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/242/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/224/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/243/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/225/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/244/)

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**